src/vs/platform/userDataSync/common/userDataSyncEnablementService.ts

```diff
@@ -52,8 +52,10 @@ export class UserDataSyncEnablementService extends Disposable implements IUserDa
 		this.storageService.store(enablementKey, enabled, StorageScope.APPLICATION, StorageTarget.MACHINE);
 	}
 
-	isResourceEnabled(resource: SyncResource): boolean {
-		return this.storageService.getBoolean(getEnablementKey(resource), StorageScope.APPLICATION, true);
+	isResourceEnabled(resource: SyncResource, defaultValue?: boolean): boolean {
+		const storedValue = this.storageService.getBoolean(getEnablementKey(resource), StorageScope.APPLICATION);
+		defaultValue = defaultValue ?? resource !== SyncResource.Prompts;
```

Comment:
I notice that the value `true` is passed in in userDataSync. Wouldn't `defaultValue` always stay `true` in this case?

Reply:
This is a change made by @sandy081 who own the code in https://github.com/microsoft/vscode/pull/242567 so cannot comment on that. This is a simply a cherry pick of that logic onto the `main` branch.